index.js

```diff
@@ -221,12 +233,18 @@ function compression (options) {
       })
 
       stream.on('end', function onStreamEnd () {
-        _end.call(res)
+        endOnce.call(res)
       })
 
       _on.call(res, 'drain', function onResponseDrain () {
         stream.resume()
       })
+
+      res.socket.on('close', function onSocketClose () {
```

Comment:
Changing this line to

```suggestion
      require('node:stream').finished(res.socket, function onFinished () {
```

works as well but it leads to a broken unit test:

```
  58 passing (173ms)
  1 pending
  1 failing

  1) compression()
       http2
         should work with http2 server:
     Uncaught Error [ERR_HTTP2_NO_SOCKET_MANIPULATION]: HTTP/2 sockets should not be directly manipulated (e.g. read and written)
      at new NodeError (node:internal/errors:405:5)
      at Object.get (node:internal/http2/compat:239:15)
      at isWritableNodeStream (node:internal/streams/utils:39:16)
      at Function.eos [as finished] (node:internal/streams/end-of-stream:78:40)
      at Http2ServerResponse.onResponseHeaders (index.js:243:30)
      at Http2ServerResponse.writeHead (node_modules/on-headers/index.js:35:16)
      at Http2ServerResponse.end (index.js:121:14)
      at /Users/martin.slota/Code/compression/test/compression.js:313:13
      at /Users/martin.slota/Code/compression/test/compression.js:1091:7
      at compression (index.js:250:5)
      at Http2Server.<anonymous> (test/compression.js:1084:5)
      at Http2Server.emit (node:events:517:28)
      at Http2Server.onServerStream (node:internal/http2/compat:917:10)
      at Http2Server.emit (node:events:517:28)
      at ServerHttp2Session.sessionOnStream (node:internal/http2/core:3010:19)
      at ServerHttp2Session.emit (node:events:517:28)
      at emit (node:internal/http2/core:330:3)
      at process.processTicksAndRejections (node:internal/process/task_queues:85:22)
```

Reply:
We really don't need to check the socket and the 'close' event, since it doesn't cover everything, that's why the 'finished' function exists.

Maybe it'll work with that.

```suggestion
      require('node:stream').finished(res, function onFinished () {
```

Reply:
All right, I'll commit the change, but I think it breaks the unit test around HTTP/2.

Reply:
Maybe it was breaking because you were accessing the socket—I'll test it locally.

Reply:
Oh, I see, you passed in `res` instead of `res.socket`. You are right, worked like a charm! 💯

Reply:
>Maybe it was breaking because you were accessing the socket—I'll test it locally.
Yes, it was - sorry about that. 🙏